### PR TITLE
[Asset][BrowserKit][Cache][Console][CssSelector] Use CPP

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | --
| License       | MIT

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
It'd be great if you could group your next PRs on the topic in a bigger one :pray:

**Comment by OskarStark** [comment]
Sure, let this one open, I can work on this PR

**Comment by OskarStark** [comment]
Can be merged please

File: src/Symfony/Component/Asset/Package.php [hunk 1]
```diff
@@ -23,12 +23,12 @@
...
     private ContextInterface $context;
 
-    public function __construct(VersionStrategyInterface $versionStrategy, ContextInterface $context = null)
-    {
-        $this->versionStrategy = $versionStrategy;
+    public function __construct(
+        private VersionStrategyInterface $versionStrategy,
+        ContextInterface $context = null,
+    ) {
         $this->context = $context ?? new NullContext();
```
Comment (wouterj) [inline src/Symfony/Component/Asset/Package.php:32]:
```suggestion
        ContextInterface $context = new NullContext(),
    ) {
```

File: src/Symfony/Component/Asset/VersionStrategy/StaticVersionStrategy.php [hunk 2]
```diff
@@ -18,16 +18,16 @@
...
      * @param string $format  Url format
      */
-    public function __construct(string $version, string $format = null)
-    {
-        $this->version = $version;
+    public function __construct(
+        private string $version,
+        string $format = null,
+    ) {
         $this->format = $format ?: '%s?%s';
```
Comment (wouterj) [inline src/Symfony/Component/Asset/VersionStrategy/StaticVersionStrategy.php:31]:
```suggestion
        private string $format = '%s?%s',
    ) {
```

File: src/Symfony/Component/BrowserKit/Request.php [hunk 3]
```diff
@@ -16,37 +16,31 @@
...
+     * @param array       $files      An array of uploaded files
+     * @param array       $cookies    An array of cookies
+     * @param array       $server     An array of server parameters
+     * @param string|null $content    The raw body data
      */
-    public function __construct(string $uri, string $method, array $parameters = [], array $files = [], array $cookies = [], array $server = [], string $content = null)
+    public function __construct(
+        private string $uri,
+        private string $method,
+        array $parameters = [],
```
Comment (nicolas-grekas) [inline src/Symfony/Component/BrowserKit/Request.php]: I missed checking that on previous PRs on the topic, but here we can declare this argument using CPP.
Can you check of that's the case is other locations where we don't use CPP but we could, even if we assign the property in the body?
(The most common case where we cannot is when the argument accepts null but the property doesn't.)

File: src/Symfony/Component/Cache/Marshaller/DefaultMarshaller.php [hunk 4]
```diff
@@ -21,17 +21,17 @@
 class DefaultMarshaller implements MarshallerInterface
 {
     private bool $useIgbinarySerialize = true;
-    private bool $throwOnSerializationFailure = false;
 
-    public function __construct(bool $useIgbinarySerialize = null, bool $throwOnSerializationFailure = false)
-    {
+    public function __construct(
+        bool $useIgbinarySerialize = null,
+        private bool $throwOnSerializationFailure = false,
```
Comment (wouterj) [inline src/Symfony/Component/Cache/Marshaller/DefaultMarshaller.php]: Should be reverted, property has a default value

File: src/Symfony/Component/Asset/Package.php (hunk 1 above)
Comment (OskarStark) [reply src/Symfony/Component/Asset/Package.php:32]: This is a BC break, as you cannot use `null` anymore

https://3v4l.org/mm35O

File: src/Symfony/Component/BrowserKit/Request.php (hunk 3 above)
Comment (OskarStark) [reply src/Symfony/Component/BrowserKit/Request.php]: Ok for now I used it like "when we do anything with the given argument, don't Use CPP. Only Use CPP if we set the exact same value to the property"

so here you want to Use CPP for `$arguments`, even we are doing the `array_walk_recursive` right?

File: src/Symfony/Component/Asset/VersionStrategy/StaticVersionStrategy.php (hunk 2 above)
Comment (OskarStark) [reply src/Symfony/Component/Asset/VersionStrategy/StaticVersionStrategy.php:31]: Same as above

File: src/Symfony/Component/Cache/Marshaller/DefaultMarshaller.php (hunk 4 above)
Comment (OskarStark) [reply src/Symfony/Component/Cache/Marshaller/DefaultMarshaller.php]: Ah you mean it has a default value on the property, right?

File: src/Symfony/Component/BrowserKit/Request.php [hunk 5]
```diff
@@ -16,37 +16,31 @@
...
+     * @param array       $parameters The request parameters
+     * @param array       $files      An array of uploaded files
+     * @param array       $cookies    An array of cookies
+     * @param array       $server     An array of server parameters
+     * @param string|null $content    The raw body data
      */
-    public function __construct(string $uri, string $method, array $parameters = [], array $files = [], array $cookies = [], array $server = [], string $content = null)
+    public function __construct(
+        private string $uri,
+        private string $method,
```
Comment (OskarStark) [inline src/Symfony/Component/BrowserKit/Request.php]:
```suggestion
        protected string $uri,
        protected string $method,
```

File: src/Symfony/Component/BrowserKit/Request.php [hunk 6]
```diff
@@ -16,37 +16,31 @@
...
+        array $parameters = [],
+        private array $files = [],
+        private array $cookies = [],
+        private array $server = [],
+        private ?string $content = null)
```
Comment (OskarStark) [inline src/Symfony/Component/BrowserKit/Request.php]:
```suggestion
        protected array $files = [],
        protected array $cookies = [],
        protected array $server = [],
        protected ?string $content = null,
        )
```

File: src/Symfony/Component/BrowserKit/Request.php (hunk 3 above)
Comment (OskarStark) [reply src/Symfony/Component/BrowserKit/Request.php]: Done ✅

File: src/Symfony/Component/Asset/Package.php (hunk 1 above)
Comment (wouterj) [reply src/Symfony/Component/Asset/Package.php:32]: Ah, I see. Good thinking!

File: src/Symfony/Component/Cache/Marshaller/DefaultMarshaller.php (hunk 4 above)
Comment (wouterj) [reply src/Symfony/Component/Cache/Marshaller/DefaultMarshaller.php]: Yes :)

File: src/Symfony/Component/Console/Formatter/OutputFormatterStyleStack.php [hunk 7]
```diff
@@ -24,11 +24,9 @@ class OutputFormatterStyleStack implements ResetInterface
      */
     private array $styles = [];
 
-    private OutputFormatterStyleInterface $emptyStyle;
-
-    public function __construct(OutputFormatterStyleInterface $emptyStyle = null)
-    {
-        $this->emptyStyle = $emptyStyle ?? new OutputFormatterStyle();
+    public function __construct(
+        private OutputFormatterStyleInterface $emptyStyle = new OutputFormatterStyle(),
```
Comment (wouterj) [inline src/Symfony/Component/Console/Formatter/OutputFormatterStyleStack.php]: We should revert this one then :)

Comment (OskarStark) [reply src/Symfony/Component/Console/Formatter/OutputFormatterStyleStack.php]: Indeed, thanks

**Comment by nicolas-grekas** [comment]
Thank you @OskarStark.

**Pull request merged**